icu4j/main/classes/core/src/com/ibm/icu/text/DateTimePatternGenerator.java

```diff
@@ -380,11 +400,16 @@ private void getAllowedHourFormats(ULocale uLocale) {
         }
 
         if (list != null) {
-            defaultHourFormatChar = list[0].charAt(0);
-            allowedHourFormats = Arrays.copyOfRange(list, 1, list.length - 1);
+            if (defaultCharFromLocale != null) {
+                defaultHourFormatChar = defaultCharFromLocale;
+                allowedHourFormats = Arrays.copyOfRange(list, 0, list.length - 1);
```

Comment:
Same comment as for C version, we should always do Arrays.copyOfRange(list, 1, list.length - 1) here